### ISSUE TITLE
Using block cursor only for normal state?

### ISSUE THREAD

**Issue by dieggsy** [fish-shell/fish-shell#4656]
|             |          | 
|-----------------|---------|
|fish version | `2.7.0`|
|uname -a | `Linux Ragnarok 4.14.12-1-ARCH #1 SMP PREEMPT Fri Jan 5 18:19:34 UTC 2018 x86_64 GNU/Linux`|
| TERM |   `xterm-termite` or `xterm-256color` (happens with both) | 
| `sh -c 'env HOME=$(mktemp -d) fish'` | behavior persists |

Basically, I use `fish_vi_key_bindings` along with:
```fish
set -U fish_cursor_insert line
set -U fish_cursor_replace_one underscore
```
However, when running a repl program like `python` (or really anything else where my cursor is visible, like `pacman -S` and other password prompting programs), the cursor I get is a block (normal mode) cursor. Is there a way for the default cursor in all instances/programs to be a line cursor, and only switch to a bar cursor when explicitly switching to normal mode?

**Comment by dieggsy** [comment]
@zanchey That makes the cursor a line in those programs, but also a line in fish's normal mode. I'd like to still have a block cursor in normal mode.

I understand fish doesn't have control over what those programs do, but them having a block cursor is a direct result of fish changing the cursor to a block - since this is a fish-specific issue.

**Comment by dieggsy** [comment]
To clarify, it would be nice to be able to do `set -U fish_cursor_default line` as well as something ilke `set -U fish_cursor_normal block`. The way it is now, normal mode (and all other programs run under fish) inherit from default cursor while insert cursor must be specified/changed. It'd be nice if the normal mode cursor could be specified or changed independently from default cursor.

**Comment by faho** [comment]
So, this would require an actual code change. What we'd need to do is to introduce a new variable that describes the cursor outside of fish. So that, when we start a command, we use that to set the cursor shape instead of `fish_cursor_default`. Any idea for the name of that variable?

- `fish_cursor_outside`

- `fish_cursor_command`

- `fish_cursor_nonfish`

- `fish_cursor_exec`

?

**Comment by dieggsy** [comment]
@faho couldn't you use fish_cursor_default for that and introduce a cursor for nomal mode? (I have no idea how this works, and you obviously know the code better, just wondering). If this isn't possible, I like `fish_cursor_exec`.

**Comment by faho** [comment]
>couldn't you use fish_cursor_default for that

Nope, because the vi "normal mode" is called "default mode" internally. So the variable name is `fish_cursor_$fish_bind_mode`, which becomes "fish_cursor_default".

>I like fish_cursor_exec.

Or how about `fish_cursor_external`?

**Comment by dieggsy** [comment]
Ah, ok. Thanks. 

> Or how about `fish_cursor_external`?

Even better :)

**Comment by zanchey** [comment]
Fixed with merge of #9565.
